linkerd/system/src/linux.rs

```diff
@@ -39,9 +42,28 @@ pub fn open_fds(pid: pid_t) -> io::Result<u64> {
 }
 
 pub fn max_fds() -> io::Result<Option<u64>> {
-    let limit = pid::limits_self()?.max_open_files;
-    let max_fds = limit.soft.or(limit.hard).map(|max| max as u64);
-    Ok(max_fds)
+    match Process::myself() {
```

Comment:
Maybe a questionable decision, but changing the signature to return `ProcResult` as we have done above might make the code easier to grok. As far as I can tell, this function has only caller so changing the signature should be safe enough, and it's a better trade off than complicating the rest of the function and maintaining an unused code path in the caller (i.e. where we handle errors).

@hawkw might have a different opinion though.

imo it would be good to just do

```suggestion
 let max_fd = Process::myself()?.limits()?.max_open_files;
```

Reply:
I agree with you that if we change the signature to return `ProcResult` the code becomes must cleaner and all the error handling can be eliminated.
I will be glad to make that change if you or @hawkw can confirm that.

Reply:
I'm fine with this --- I believe `io::Result` is the return type here only because the `procinfo` API returns an `io::Result`. It's fine to change this.